ntrans_dataprep.py

```diff
@@ -37,24 +51,25 @@ def write_data_to_csv(n_to_ngrams):
                 data_writer.writerow(csv_row)
 
 
-def count_ngram_frequency(n_to_ngrams):
+def count_ngram_frequency(n_to_ngrams, data_chunk):
```

Comment:
This function should be renamed. It writes a file, but it's not at all apparent from the named. Maybe just combine this with the `write_data_to_csv` function, since `write_data_to_csv` is never called anywhere else than inside this function?

Reply:
Hmm, I'm not sure I agree here. This function counts the N-grams and calls the function that writes the files. 

I am hesitant to combine the two functions as I keep reading that a function should "do one thing", rather than having different sections. 

I don't think it adds confusion the way it is.

Reply:
Maybe like this then?

```python
def count_ngram_frequency(n_to_ngrams):
    return {n: collections.Counter(ngrams) for n, ngrams in n_to_ngrams.items()}
```

The function name hasn't changed. It's still `count_ngram_frequency()`, but it no longer does two things: instead of creating counters and writing to a file, it just creates counters.

Then writing to file would look like this:

```python
write_data_to_csv(count_ngram_frequency(n_to_ngrams), data_chunk)
```

Reply:
This looks great! :)